packages/graphql/src/modules/user/app/authenticateFacebookUser.spec.ts

```diff
@@ -0,0 +1,197 @@
+import User from '../domain/User';
+import createFakeStorageProvider from '../../../shared/providers/StorageProvider/fakes/fakeStorageProvider';
+import createFakeUsersRepository from '../infra/fakes/fakeUsersRepository';
+import createFakeSocialMediaRepository from '../infra/fakes/fakeSocialMediaRepository';
+import createFakeTokenProvider from '../providers/TokenProvider/fakes/fakeTokenProvider';
+
+import authenticateFacebookUser from './authenticateFacebookUser';
+
+const authenticateUserParams = {
+  token: 'fake-token',
+  permissions: ['public_profile', 'email'],
+  expires: 60 * 60 * 24 * 30,
+};
+
+const fakeUser = new User({
+  id: 'fake-uuid',
+  name: 'fake-user',
+  emails: [
+    {
+      email: 'mysocialemail@fake.com',
+      providedBy: 'facebook',
+    },
+  ],
+  profileImages: [
+    {
+      providedBy: 'facebook',
+      url: 'fake-token/profilePicture.jpeg',
+    },
+  ],
+  providersIds: [
+    {
+      id: 'socialId-fake-token',
+      providedBy: 'facebook',
+    },
+  ],
+});
+
+const setUp = () => {
+  const userRepository = createFakeUsersRepository();
+  const storageProvider = createFakeStorageProvider();
+  const facebookRepository = createFakeSocialMediaRepository();
+  const tokenProvider = createFakeTokenProvider({userRepository});
+  const authenticateUser = authenticateFacebookUser({
+    facebookRepository,
+    storageProvider,
+    tokenProvider,
+    userRepository,
+  });
+
+  return {
+    authenticateUser,
+    storageProvider,
+    userRepository,
+    facebookRepository,
+    tokenProvider,
+  };
+};
+
+describe('AuthenticateFacebookUser', () => {
+  it('should be able to authenticate', async () => {
```

Comment:
Not a big thing but tests shouldn't start with `should` they need to be assertive about what they are doing
https://www.betterspecs.org/#should

Reply:
Cool, I didn't know that I will address this issue

Reply:
Pretty cool this site https://www.betterspecs.org/ by the way!